component/http/component.go

```diff
@@ -56,6 +56,10 @@ func (c *Component) Run(ctx context.Context) error {
 		log.Info("shutting down component")
 		return srv.Shutdown(ctx)
 	case err := <-chFail:
+		log.Errorf("shutting down component due to error: %v", err)
```

Comment:
This changes the behaviour of the existing HTTP server. Did we miss something? a bug?

Reply:
yes, i noticed during a test, that actually in the case of error, the component would not shut down. Would you prefer to have it as a separate issue ? It was just blocking the cache tests ...

Reply:
http should not shutdown in case of an error in a request. we handle the error (metrics,logs eventually) and go to the next request.
If a panic occurs we recover and continue. so in general http server will stay up for everything.

Reply:
ok, fair point. It was added for the test case `TestComponent_ListenAndServeTLS_FailsInvalidCerts`. Which i thought should also shutdown the server. But i have probably missed the bigger contet.
Reverted now.